src/buffer.rs

```diff
@@ -355,6 +529,15 @@ where
         }
     }
 
+    /// Returns an iterator over the rows of this image.
+    pub fn rows(&self) -> Rows<P> {
+        Rows {
+            chunks: self
+                .data
+                .chunks((<P as Pixel>::channel_count() as u32 * self.width) as usize),
```

Comment:
This multiplication needs to happen with type `usize`. Otherwise the resulting chunks will be too small on some images (Note that `u64` is not necessary as the complete image can be addressed).